packages/xo-server/src/xo-mixins/pool.js

```diff
@@ -0,0 +1,93 @@
+import { difference } from 'lodash'
+
+export default class {
+  constructor(xo) {
+    this._xo = xo
+  }
+
+  async mergeInto({ sources, target, force }) {
+    const targetHost = this._xo.getObject(target.master)
+    const _sources = sources.map(sourceId =>
+      this._xo.getObject(sourceId, 'pool')
+    )
+    let source
+    let sourceHost
+    const patchesSources = {}
+    for (source of _sources) {
+      sourceHost = this._xo.getObject(source.master)
+      if (sourceHost.productBrand !== targetHost.productBrand) {
+        throw new Error(
+          `a ${sourceHost.productBrand} pool cannot be merged into a ${
+            targetHost.productBrand
+          } pool`
+        )
+      }
+      patchesSources[source.id] = sourceHost.patches.map(
+        patchId => this._xo.getObject(patchId).name
+      )
+    }
+
+    const requiredPatchesSources = {}
+    let requiredPatchesTarget = []
+    for (source of _sources) {
+      requiredPatchesTarget = requiredPatchesTarget.concat(
+        this._xo.getPatchesDifference(source.master, target.master)
+      )
+      requiredPatchesSources[source.id] = this._xo.getPatchesDifference(
+        target.master,
+        source.master
+      )
+    }
+
+    requiredPatchesTarget = [...new Set(requiredPatchesTarget)]
+    const newPatchesTarget = requiredPatchesTarget.concat(
+      targetHost.patches.map(patchId => this._xo.getObject(patchId).name)
+    )
+    let newPatchesSource
+    let requiredPatchesSource
+    for (source of _sources) {
+      requiredPatchesSource = requiredPatchesSources[source.id]
+      newPatchesSource = requiredPatchesSource.concat(patchesSources[source.id])
+      requiredPatchesTarget = requiredPatchesTarget.concat(
+        difference(newPatchesSource, newPatchesTarget)
+      )
```

Comment:
I'm not sure why you're doing this. At this point, you should already know which patches must be installed on the target, right? Only the sources miss some information.